rotkehlchen/exchanges/kraken.py

```diff
@@ -782,3 +779,282 @@ def query_online_income_loss_expense(
+                                f'will be stored but not processed.',
+                            )
+                        fee = deserialize_asset_amount(raw_event['fee'])
+
+                        group_events.append(HistoryEvent(
+                            event_identifier=identifier,
+                            sequence_index=idx,
+                            timestamp=timestamp,
+                            location=Location.KRAKEN,
+                            location_label=self.name,
+                            amount=AssetBalance(
+                                asset=asset,
+                                balance=Balance(
+                                    amount=amount,
+                                    usd_value=ZERO,
+                                ),
+                            ),
+                            notes=None,
+                            event_type=event_type,
+                            event_subtype=event_subtype,
+                        ))
+                        if fee != ZERO:
+                            group_events.append(HistoryEvent(
+                                event_identifier=identifier,
+                                sequence_index=len(events),
+                                timestamp=timestamp,
+                                location=Location.KRAKEN,
+                                location_label=self.name,
+                                amount=AssetBalance(
+                                    asset=asset,
+                                    balance=Balance(
+                                        amount=fee,
+                                        usd_value=ZERO,
+                                    ),
+                                ),
+                                notes=None,
+                                event_type=event_type,
+                                event_subtype=HistoryEventSubType.FEE,
+                            ))
+                    except (DeserializationError, KeyError, UnknownAsset) as e:
+                        msg = str(e)
+                        if isinstance(e, KeyError):
+                            msg = f'Keyrror {msg}'
+                        self.msg_aggregator.add_error(
+                            f'Failed to read ledger event from kraken {raw_event} due to {msg}',
+                        )
+                        break
+                else:
+                    # In the case where a part of the event can't be correctly procesed change
+                    # the event type of the whole group
+                    if found_unknown_event:
+                        for event in group_events:
+                            event.event_type = HistoryEventType.UNKNOWN
+                    new_events.extend(group_events)
+            # Make sure that events are saved and if we fail to save them don't update
+            # the time ranges queried
+            is_db_saved = True
```

Comment:
As I mentioned in my comment in `save_history_events()` it should not really fail as it stands. May omit a write of 1 event at deserialization but that's it. So this logic can go away.